src/arbitraries/MappedArbitrary.ts

```diff
@@ -32,8 +39,17 @@ export class MappedArbitrary<A, B> extends Arbitrary<B> {
   }
 
   canGenerate(pick: FluentPick<B>) {
-    /* && pick.value === this.f(pick.original) */
-    return this.baseArbitrary.canGenerate({value: pick.original, original: pick.original})
+    const inverseValues = !(this.config && this.config.inverseFunction) ? [pick.original] :
+      (Array.isArray(this.config.inverseFunction(pick.value)) ?
+        this.config.inverseFunction(pick.value) as A[] :
+        [this.config.inverseFunction(pick.value)])
```

Comment:
Would this work for arrays generated by an `ArbitraryArray`?

Reply:
I also have this doubt. Why do you need to have these two cases?

Reply:
The code here being discussed has now changed (make sure you check the new modifications). Despite the alterations, I have added a new test using `ArbitraryArrays` and `MappedAribtrary` to make sure everything is working.